### ISSUE TITLE
[Bug]: Dropdown / Combobox: The hover state text colour of the drop down is not accessible

### ISSUE THREAD

**Issue by jurokapsiar** [microsoft/fluentui#29367]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Teams HC Theme
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://react.fluentui.dev/?path=/docs/components-dropdown--default&globals=storybook_fluentui-react-addon_theme:teams-high-contrast

## Actual Behavior
<img width="333" alt="image" src="https://github.com/microsoft/fluentui/assets/13742664/a30be47e-6cac-4978-86a5-d4167614950a">

## Expected Behavior

As deifned in the spec:
<img width="655" alt="image" src="https://github.com/microsoft/fluentui/assets/13742664/4d153745-16d3-47d6-ab84-770fdd486d24">

tokens suggested by Daisy:
<img width="1269" alt="image" src="https://github.com/microsoft/fluentui/assets/13742664/2ebabc8e-47a2-4380-a874-5dfc51dd3a13">

### Requested priority

Blocking

### Products/sites affected

Teams

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
